src/plugins/kibana_legacy/public/utils/system_api.ts

```diff
@@ -44,5 +45,7 @@ export function addSystemApiHeader(originalHeaders: Record<string, string>) {
  */
 export function isSystemApiRequest(request: IRequestConfig) {
   const { headers } = request;
-  return headers && !!headers[SYSTEM_API_HEADER_NAME];
+  return (
+    headers && (!!headers[SYSTEM_REQUEST_HEADER_NAME] || !!headers[LEGACY_SYSTEM_API_HEADER_NAME])
+  );
 }
```

Comment:
This is still being used by `x-pack/legacy/plugins/security/public/hacks/on_session_timeout.js`, but this will be removed by 8.0.

Do we need to add a deprecation tag to this?

Reply:
I don't think a deprecation tag is required here, it's residing within the `kibana_legacy` plugin, so all of this code is technically deprecated.